Automattic/wp-calypso
# Replace the StatsWriter plugin with better performing BundleAnalyzerPlugin

For generating the `stats.json` file, remove our own `StatsWriter` that uses `JSON.stringify` and replace it with `BundleAnalyzerPlugin` that uses the `bfj` library to write the JSON file as a stream, consuming less memory, and is built into the `webpack-bundle-analyzer` NPM package.

Also, webpack doesn't need the `profile` flag when emitting stats: we don't use the timing information there when analyzing the bundles.

This change should reduce webpack memory requirements when generating the stats.

#### Testing instructions

Verify that `npm run analyze-bundes` still works as expected. ICFY on this branch reports "No changes in production JS", which is a good sign :wink:

Can we somehow verify what is the memory consumption of the CircleCI `icfy-stats` job before and after this PR?
